HelloVectors/HelloVectors_win32_cpp/LottieTool/SquareCircleMorph.cpp

```diff
@@ -0,0 +1,355 @@
+//------------------------------------------------------------------------------
+// <auto-generated>
+//     This code was generated by a tool.
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
+#include "pch.h"
+#include "SquareCircleMorph.h"
+#include "d2d1.h"
+#include <d2d1_1.h>
+#include <d2d1helper.h>
+
+#include <Windows.Graphics.Interop.h>
+
+using namespace winrt;
+using namespace Windows::UI::Composition;
+using namespace Windows::Graphics;
+
+namespace
+{
+	class GeoSource final :
+		public ABI::Windows::Graphics::IGeometrySource2D,
+		public ABI::Windows::Graphics::IGeometrySource2DInterop
+	{
+		ULONG _cRef;
+		com_ptr<ID2D1Geometry> _cpGeometry;
+
+	public:
+		GeoSource(com_ptr<ID2D1Geometry> pGeometry)
+			: _cRef(1)
+			, _cpGeometry(pGeometry)
+		{ }
+
+		IFACEMETHODIMP QueryInterface(REFIID iid, void ** ppvObject) override
```

Comment:
This QI implementation is faulty. Just use winrt::implements and get a correct implementation for free.

Reply:
Fixed in next PR